python/ray/serve/_private/router.py

```diff
@@ -160,9 +171,38 @@ def _all_query_refs(self):
 
     def _drain_completed_object_refs(self) -> int:
         refs = self._all_query_refs
+        # NOTE(simon): even though the timeout is 0, a large number of refs can still
```

Comment:
Curious do you observe the performance regression here? and how many refs causing it?

Reply:
No major regression, just an observation from @iycheng

Reply:
Yes, I make it very big and in case of a failure, somehow the through put drop slowly and we believe it's because of this. And ray.wait doesn't come free.